src/object/schema_template.c

```diff
@@ -699,13 +699,14 @@ def_class_internal (const char *name, int class_type)
   char realname[SM_MAX_IDENTIFIER_LENGTH];
   SM_TEMPLATE *template_ = NULL;
   PR_TYPE *type;
+  const char *simple_name = sm_simple_name (name);
 
   if (sm_check_name (name))
     {
-      type = pr_find_type (name);
+      type = pr_find_type (simple_name);
```

Comment:
if (sm_check_name (name))
    {
      const char *simple_name = sm_simple_name (name);
      type = pr_find_type (simple_name);

Reply:
Okay, I'll change it.